DataFormats/PatCandidates/src/PackedCandidate.cc

```diff
@@ -62,9 +62,12 @@ void pat::PackedCandidate::packVtx(bool unpackAfterwards) {
 }
 
 void pat::PackedCandidate::unpack() const {
-    p4_ = PolarLorentzVector(MiniFloatConverter::float16to32(packedPt_),
+    float pt = MiniFloatConverter::float16to32(packedPt_);
+    double shift = (pt<1. ? 0.1*pt : 0.1/pt); // shift particle phi to break degeneracies in angular separations
+    double phi = int16_t(packedPhi_)*3.2f/std::numeric_limits<int16_t>::max() + shift*3.2/std::numeric_limits<int16_t>::max();
```

Comment:
at constr we do pack() + unpack() to be sure that transient packed already
contains the approx.
did you check if this op is stable?
Il 11/giu/2015 19:06, "Dinko Ferencek" <notifications@github.com> ha
scritto:

> In DataFormats/PatCandidates/src/PackedCandidate.cc
> <https://github.com/cms-sw/cmssw/pull/9554#discussion_r32243557>:
>
> > @@ -62,9 +62,12 @@ void pat::PackedCandidate::packVtx(bool unpackAfterwards) {
> >  }
> >
> >  void pat::PackedCandidate::unpack() const {
> > -    p4_ = PolarLorentzVector(MiniFloatConverter::float16to32(packedPt_),
> > +    float pt = MiniFloatConverter::float16to32(packedPt_);
> > +    double shift = (pt<1. ? 0.1*pt : 0.1/pt); // shift particle phi to break degeneracies in angular separations
> > +    double phi = int16_t(packedPhi_)*3.2f/std::numeric_limits<int16_t>::max() + shift*3.2/std::numeric_limits<int16_t>::max();
>
> By construction, the degridification cannot be done at the packing stage,
> since it is the packing that introduces the grid (the packing process
> involves mapping to integers so there are no fractional integers
> available). The degridification can only be introduced at the unpacking
> stage.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/9554/files#r32243557>.
>